tikv-client/src/main/java/com/pingcap/tikv/KVClient.java

```diff
@@ -0,0 +1,234 @@
+/*
+ *
+ * Copyright 2019 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.pingcap.tikv;
+
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.exception.GrpcException;
+import com.pingcap.tikv.exception.TiKVException;
+import com.pingcap.tikv.operation.iterator.ConcreteScanIterator;
+import com.pingcap.tikv.region.RegionStoreClient;
+import com.pingcap.tikv.region.RegionStoreClient.*;
+import com.pingcap.tikv.region.TiRegion;
+import com.pingcap.tikv.util.BackOffFunction;
+import com.pingcap.tikv.util.BackOffer;
+import com.pingcap.tikv.util.ConcreteBackOffer;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+import org.apache.log4j.Logger;
+import org.tikv.kvproto.Kvrpcpb;
+import org.tikv.kvproto.Kvrpcpb.KvPair;
+
+public class KVClient implements AutoCloseable {
+  private final RegionStoreClientBuilder clientBuilder;
+  private final TiConfiguration conf;
+  private final ExecutorService executorService;
+  private static final Logger logger = Logger.getLogger(KVClient.class);
+
+  private static final int BATCH_GET_SIZE = 16 * 1024;
+
+  public KVClient(TiConfiguration conf, RegionStoreClientBuilder clientBuilder) {
+    Objects.requireNonNull(conf, "conf is null");
+    Objects.requireNonNull(clientBuilder, "clientBuilder is null");
+    this.conf = conf;
+    this.clientBuilder = clientBuilder;
+    // TODO: ExecutorService executors =
+    // Executors.newFixedThreadPool(conf.getKVClientConcurrency());
+    executorService = Executors.newFixedThreadPool(20);
+  }
+
+  @Override
+  public void close() {
+    if (executorService != null) {
+      executorService.shutdownNow();
+    }
+  }
+
+  /**
+   * Get a raw key-value pair from TiKV if key exists
+   *
+   * @param key raw key
+   * @return a ByteString value if key exists, ByteString.EMPTY if key does not exist
+   */
+  public ByteString get(ByteString key, long version) throws GrpcException {
```

Comment:
missed BackOffTimeOut Exception?

Reply:
when backoff timeout, it throws an GrpcException